src/fsharp/CompileOps.fs

```diff
@@ -4879,27 +4886,34 @@ let ProcessMetaCommandsFromInput
                List.fold (fun state d -> nowarnF state (m,d)) state numbers
 
             | ParsedHashDirective(("reference" | "r"),args,m) -> 
-               if not canHaveScriptMetaCommands then
-                   errorR(HashReferenceNotAllowedInNonScript m)
+                if not canHaveScriptMetaCommands then
+                    errorR(HashReferenceNotAllowedInNonScript m)
+
+                let reportError errorType error =
+                    match errorType with
+                    | ErrorReportType.Warning -> warning(Error(error,m))
+                    | ErrorReportType.Error -> errorR(Error(error, m))
+
+                match args with
+                | [path] ->
+                    matchedm <- m
+                    let dm = tcConfig.dependencyProvider.TryFindDependencyManagerInPath(tcConfig.compilerToolPaths, tcConfig.outputDir |> Option.defaultValue "" , reportError, path)
+                    match dm with
+                    | dllpath, null when String.IsNullOrEmpty(dllpath) ->
```

Comment:
`IsNullOrEmpty` --> what if it's just whitespace?

Reply:
@cartermp that is a typo, my brain said IsNullOrWhitespace my fingers went with IsNullOrEmpty.  I need new fingers.